Celeste.Mod.mm/Patches/UserIO.cs

```diff
@@ -54,6 +55,13 @@ public static string GetSaveFilePath(string name = null)
             if (Saving)
                 QueuedSaves.Enqueue(Tuple.Create(file, settings));
 
+            if (settings && UserIO.Open(UserIO.Mode.Write)) {
+                // VanillaMouseBindings extracts the data from the Settings class
+                byte[] data = UserIO.Serialize(new VanillaMouseBindings().Init());
+                UserIO.Save<VanillaMouseBindings>("modsettings-Everest_MouseBindings", data);
+                UserIO.Close();
+            }
+
```

Comment:
If you save settings here it bypasses the save thread and any error handling. Might make more sense to add the serialization in `SaveRoutine` and do the saving in `orig_SaveThread` next to where it does that for vanilla settings?

Reply:
Serializing and Saving have been separated, with saving happening in the `SaveThread`, and mirrors how Mod SaveData is handled.